Code/Framework/AzToolsFramework/Tests/Prefab/PrefabCreateInMemoryTests.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+#include <Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    using PrefabCreateInMemoryTest = PrefabTestFixture;
+
+    TEST_F(PrefabCreateInMemoryTest, CreatePrefabFromEntitiesAndValidateChildEntityOrder)
+    {
+        // Level
+        // | Car
+        //   | Engine   (entity)
+        //   | Wheel    (instance)
+        //     | Tire
+        //   | Battery  (entity)
+
+        const AZStd::string carPrefabName = "CarPrefab";
+        const AZStd::string wheelPrefabName = "WheelPrefab";
+        const AZStd::string tireEntityName = "Tire";
+        const AZStd::string engineEntityName = "Engine";
+        const AZStd::string batteryEntityName = "Battery";
+
+        AZ::IO::Path engineRootPath;
+        m_settingsRegistryInterface->Get(engineRootPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder);
+        AZ::IO::Path carPrefabFilepath = engineRootPath / carPrefabName;
+        AZ::IO::Path wheelPrefabFilepath = engineRootPath / wheelPrefabName;
+
+        // Create an Engine entity first
+        AZ::EntityId engineEntityId = CreateEditorEntityUnderRoot(engineEntityName);
+        // Create the Wheel instance after the Engine entity
+        AZ::EntityId tireEntityId = CreateEditorEntityUnderRoot(tireEntityName);
+        AZ::EntityId wheelContainerId = CreateEditorPrefab(wheelPrefabFilepath, { tireEntityId });
+        // Create the Battery entity after the Wheel instance
+        AZ::EntityId batteryEntityId = CreateEditorEntityUnderRoot(batteryEntityName);
+        // Create a Car prefab from the entities and isntance
+        AZ::EntityId carContainerId = CreateEditorPrefab(carPrefabFilepath, { engineEntityId, batteryEntityId, wheelContainerId });
+        auto carInstance = m_instanceEntityMapperInterface->FindOwningInstance(carContainerId);
+        ASSERT_TRUE(carInstance.has_value());
+        ASSERT_TRUE(carInstance->get().GetContainerEntityId() == carContainerId);
+
+        // Validate the child entity order of the car instance
+        AzToolsFramework::EntityOrderArray entityOrderArray = AzToolsFramework::GetEntityChildOrder(carContainerId);
+        EXPECT_TRUE(entityOrderArray.size() == 3);
+        AZStd::string entityName;
+        AZ::ComponentApplicationBus::BroadcastResult(entityName, &AZ::ComponentApplicationRequests::GetEntityName, entityOrderArray[0]);
```

Comment:
Learn new thing every day! I would probably do something similar to validate order in "detach prefab". Currently, the order is not covered in auto and unit tests.

The existing approach seems better. Just curious if we can also retrieve the sort order component and do the validation?

Reply:
Yes, we could also do the validation using the sort order component, but since there may be changes coming to the sort order component soon, I decided to use this API that's less likely to change.